.github/workflows/tests.yml

```
@@ -151,58 +152,99 @@ jobs:
         python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
-        test-type: ['unit', 'integration']
-        test-group: ['1', '2', '3']
+        default-channel: [defaults, conda-forge]
+        test-type: [unit, integration]
+        test-group: [1, 2, 3]
         exclude:
-          - default-channel: 'conda-forge'
-            python-version: '3.8'
-          - default-channel: 'conda-forge'
-            python-version: '3.9'
-          - default-channel: 'conda-forge'
-            python-version: '3.10'
-          - default-channel: 'conda-forge'
-            python-version: '3.11'
+          - python-version: '3.8'
+            default-channel: conda-forge
+          - python-version: '3.9'
+            default-channel: conda-forge
+          - python-version: '3.10'
+            default-channel: conda-forge
+          - python-version: '3.11'
+            default-channel: conda-forge
+          - test-type: unit
+            test-group: 3
     env:
-      OS: Linux
-      PYTHON: ${{ matrix.python-version }}
-      TEST_SPLITS: 3
-      TEST_GROUP: ${{ matrix.test-group }}
+      PYTEST_MARKER: ${{ matrix.test-type == 'unit' && 'not integration' || 'integration' }}
+      PYTEST_SPLITS: ${{ matrix.test-type == 'unit' && '2' || '3' }}
+      REQUIREMENTS_TRUSTSTORE: ${{ contains('3.10|3.11|3.12', matrix.python-version) && '--file tests/requirements-truststore.txt' || '' }}
+
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout Source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Python ${{ matrix.python-version }} on ${{ matrix.default-channel }}, ${{ matrix.test-type }} tests, group ${{ matrix.test-group }}
-        run: >
-          docker run
-          --rm -v ${PWD}:/opt/conda-src
-          -e TEST_SPLITS
-          -e TEST_GROUP
-          ghcr.io/conda/conda-ci:main-linux-python${{ matrix.python-version }}${{ matrix.default-channel == 'conda-forge' && '-conda-forge' || '' }}
-          /opt/conda-src/dev/linux/${{ matrix.test-type }}.sh
+      - name: Hash + Timestamp
+        run: echo "HASH=${{ runner.os }}-${{ runner.arch }}-Py${{ matrix.python-version }}-${{ matrix.default-channel }}-${{ matrix.test-type }}-${{ matrix.test-group }}-$(date -u "+%Y%m")" >> $GITHUB_ENV
 
-      - uses: codecov/codecov-action@v4
+      - name: Cache Conda
+        uses: actions/cache@v4
         with:
-          flags: ${{ matrix.test-type }},${{ runner.os }},${{ matrix.python-version }}
+          path: ~/conda_pkgs_dir
+          key: cache-${{ env.HASH }}
 
-      - name: Upload test results
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          auto-update-conda: true
+          conda-solver: libmamba
+          condarc-file: .github/condarc-${{ matrix.default-channel }}
+          python-version: ${{ matrix.python-version }}
+          run-post: false  # skip post cleanup
```

```suggestion
        with:
          condarc-file: .github/condarc-${{ matrix.default-channel }}
          run-post: false  # skip post cleanup
```

These options are not used in the other CI entries. Also, since we are running `conda install` next, we can save one solve and add `python=${{ matrix.python-version }}` instead there.